test/src/unit-regression2.cpp

```diff
@@ -48,6 +48,7 @@ using ordered_json = nlohmann::ordered_json;
 #endif
 
 #if JSON_HAS_EXPERIMENTAL_FILESYSTEM
+// JSON_HAS_CPP_17 (magic keyword; do not remove)
```

Comment:
This is unrelated but I want to make sure these test are built with the correct standard version and the CMake code only checks for the presence of `JSON_HAS_CPP_*` macros. `JSON_HAS_CPP_17` is mentioned right above but if this file is ever refactored, that might change.

Reply:
Please add a comment that explains this "magic", e.g.

> cmake/test.cmake expects C++-17 tests to contain JSON_HAS_CPP_17 keyword

or so.

Reply:
Alternatively, make the requirement explicit and add

```cpp
#ifdef JSON_HAS_CPP_17
```

before

```cpp
#if JSON_HAS_EXPERIMENTAL_FILESYSTEM
```

Reply:
I don't want to clutter the body of the tests with repeated explanations. Ideally every usage of a macro linked to a standard version (e.g., `JSON_HAS_RANGES`, `JSON_HAS_THREE_WAY_COMPARISON`, and probably more with each new standard) should also have that keyword close by, to avoid a repeat of tests not being built for the right standard.

I suggest a detailed note at the top of each test and a short referral to that note at each point of use. This CMake requirement is not documented or explained anywhere else right now.

```
// cmake/test.cmake selects the C++ standard versions with which to build the
// unit tests based on the presence of JSON_HAS_CPP_<VERSION> macros.
// When using macros that are only defined for particular versions of the standard
// (e.g., JSON_HAS_FILESYSTEM for C++17 and up), please mention the corresponding
// version macro in a comment close by, like this:
// JSON_HAS_CPP_<VERSION> (do not remove; see note at top of file)
```

I'd make these changes in one of the other PRs (C++20 fixes probably, since I'm using similar language there) and update all unit tests in one go. Does that work for you?